src/connection_pool.rs

```diff
@@ -90,11 +95,12 @@ where
 
         // Spawn a thread to run the resolver
         let barrier_clone = barrier.clone();
+        let tx_clone = tx.clone();
```

Comment:
Nit: the names "tx" and "rx" here are a little hard to understand in context; you might consider renaming with a little more context ("rslv_tx" or something).